src/promise-btn.directive.ts

```diff
@@ -137,15 +147,11 @@ export class PromiseBtnDirective implements OnDestroy, AfterContentInit {
   /**
    * Initializes a watcher for the promise. Also takes
    * this.cfg.minDuration into account if given.
-   * @param {Object}promise
    * @param {Object}btnEl
    */
 
-  initPromiseHandler(promise: any, btnEl: HTMLElement) {
-    // return if something else then a promise is passed
-    if (!promise || !promise.then) {
-      return;
-    }
```

Comment:
Only `checkAndInitPromiseHandler` calls `initPromiseHandler`, and, as we can see the `this.promise` variable was already checked in `checkAndInitPromiseHandler`, so check in `initPromiseHandler` is redundant and can be deleted.